biorxiv/journal_tracker/journal_recommendation_experiment.py

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Journal Recommendation for Preprints
+
+# The goal of this notebook is to help users know which journal would be most appropriate for their preprint. The central idea is to use euclidean distance between documents to gauge which journal similar works have been sent.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+from collections import Counter, defaultdict
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from sklearn.dummy import DummyClassifier
+from sklearn.neighbors import KNeighborsClassifier
+from tqdm import tqdm_notebook
+
+from annorxiver_modules.journal_rec_helper import (
+    cross_validation, dummy_evaluate,
+    knn_evaluate, knn_centroid_evaluate
+)
+
+
+# # Load bioRxiv Document Vectors
+
+# In[2]:
+
+
+biorxiv_journal_df = (
+    pd.read_csv("output/mapped_published_doi.tsv", sep="\t")
+    .groupby("doi")
+    .agg({
+        "document":"last",
+        "category":"first",
+        "journal":"first",
+        "doi":"last",
+        "published_doi":"first",  
+        "pmcid":"first", 
+    })
+    .reset_index(drop=True)
+)
+biorxiv_journal_df.head()
+
+
+# In[3]:
+
+
+# Count number of Non-NaN elements
+print(f"Number of Non-NaN entries: {biorxiv_journal_df.pmcid.count()}")
+print(f"Total number of entries: {biorxiv_journal_df.shape[0]}")
+print(f"Percent Covered: {biorxiv_journal_df.pmcid.count()/biorxiv_journal_df.shape[0]:.2f}")
+
+
+# In[4]:
+
+
+golden_set_df = biorxiv_journal_df.query("pmcid.notnull()")
+golden_set_df.head()
+
+
+# # Load Pubmed Central Document Vectors
+
+# In[5]:
+
+
+pmc_articles_df = (
+    pd.read_csv(
+        "../../pmc/exploratory_data_analysis/output/pubmed_central_journal_paper_map.tsv.xz", 
+        sep="\t"
+    )
+    .query("article_type=='research-article'")
+)
+print(pmc_articles_df.shape)
+pmc_articles_df.head()
+
+
+# In[6]:
+
+
+journals = pmc_articles_df.journal.value_counts()
+print(journals.shape)
+journals
+
+
+# In[7]:
+
+
+# Filter out low count journals
+pmc_articles_df = pmc_articles_df.query(f"journal in {journals[journals > 100].index.tolist()}")
+print(pmc_articles_df.shape)
+pmc_articles_df.head()
+
+
+# In[8]:
+
+
+pmc_embedding_dict = {
+    int(path.stem[-7:-4]):pd.read_csv(
```

Comment:
This is kind of fragile, do you know that you'll always have exactly three digit identifiers that are always in the -7 to -4 position? Might it make more sense to just use a delimiter in the file names?